pkg/sfu/streamallocator.go

```diff
@@ -617,21 +617,20 @@ func (s *StreamAllocator) runStateMachine(event Event) {
 	}
 }
 
-// LK-TODO-START
-// Signal_ADD_TRACK is not useful. Probably can get rid of it.
-// AVAILABLE_LAYERS_ADD/REMOVE should be how track start should
-// be getting an allocation.
-// LK-TODO-END
+// LK-TODO: ADD_TRACK should not reallocate if added track is audio
 func (s *StreamAllocator) runStatePreCommit(event Event) {
 	switch event.Signal {
 	case Signal_ADD_TRACK:
-		// wait for layers add/remove signal
+		s.allocate()
```

Comment:
This was not as expected because the `AddTrack` here comes after layers add/remove happens when a new subscriber joins. So, do an allocate here.